src/vs/workbench/contrib/notebook/browser/contrib/find/findMatchDecorationModel.ts

```diff
@@ -25,68 +26,91 @@ export class FindMatchDecorationModel extends Disposable {
 		return this._currentMatchDecorations;
 	}
 
-	public async highlightCurrentFindMatchDecoration(cell: ICellViewModel, match: FindMatch | CellWebviewFindMatch): Promise<number | null> {
+	set notebookEditor(notebookEditor: INotebookEditor | undefined) {
+		if (this._notebookEditor) {
+			// clear any previous decorations if applicable
+			this.clearDecorations();
+		}
+		this._notebookEditor = notebookEditor;
+	}
 
-		if (match instanceof FindMatch) {
-			this.clearCurrentFindMatchDecoration();
+	public clearDecorations() {
+		this.clearCurrentFindMatchDecoration();
+		this.setAllFindMatchesDecorations([]);
+	}
 
-			// match is an editor FindMatch, we update find match decoration in the editor
-			// we will highlight the match in the webview
-			this._notebookEditor.changeModelDecorations(accessor => {
-				const findMatchesOptions: ModelDecorationOptions = FindDecorations._CURRENT_FIND_MATCH_DECORATION;
-
-				const decorations: IModelDeltaDecoration[] = [
-					{ range: match.range, options: findMatchesOptions }
-				];
-				const deltaDecoration: ICellModelDeltaDecorations = {
-					ownerId: cell.handle,
-					decorations: decorations
-				};
-
-				this._currentMatchDecorations = {
-					kind: 'input',
-					decorations: accessor.deltaDecorations(this._currentMatchDecorations?.kind === 'input' ? this._currentMatchDecorations.decorations : [], [deltaDecoration])
-				};
-			});
-
-			this._currentMatchCellDecorations = this._notebookEditor.deltaCellDecorations(this._currentMatchCellDecorations, [{
-				ownerId: cell.handle,
-				handle: cell.handle,
-				options: {
-					overviewRuler: {
-						color: overviewRulerSelectionHighlightForeground,
-						modelRanges: [match.range],
-						includeOutput: false,
-						position: NotebookOverviewRulerLane.Center
-					}
-				}
-			} as INotebookDeltaDecoration]);
 
+	public async highlightCurrentFindMatchDecorationInCell(cell: ICellViewModel, cellRange: Range): Promise<number | null> {
```

Comment:
I split Find Match and Webview Match highlight into different functions, since search needs this. Caller is responsible for calling the correct function.